### ISSUE TITLE
Django DataError with the remote_addr field when attempting to save it in Postgresql

### ISSUE THREAD

**Issue by rrguedes** [jazzband/django-auditlog#414]
Hello,

I'm facing a strange error when saving an admin model in Postgresql. After some research, I realized it has something to do with the auditlog. The exception is being generated because Django does not recognize the GenericIPAddressField (inet) type as correct. I've checked and my server is sending both X-Forwarded-For and REMOTE-ADDR METAs to the API, but as in auditlog/middleware.py X-Forwarded-For field is got if not empty. The error doesn't happen at my local machine, nor to me once I deploy it for an Azure App Service. But strangely it happens for the clients using it.

I'm using the automatic log approach and all I did was install auditlog and register my Models.

![image](https://user-images.githubusercontent.com/2847423/184443948-1242ef42-7d49-42a3-90ce-ed781c5a7fd6.png)

I believe the problem is being caused by the fact that the current value ends with ":" and the split operation in auditlog/middleware.py considers a comma "," for splitting purposes. The way I see it, and I might be wrong, is to include an alternative split method considering ":" for cases when we do have ports in X-Forwarded-For value. Does it make sense?

![image](https://user-images.githubusercontent.com/2847423/184443324-362c3097-55e8-4a08-89c2-3af2b5eadef4.png)

Thanks.

**Comment by hramezani** [comment]
Thanks @rrguedes for reporting the issue.

> I believe the problem is being caused by the fact that the current value ends with ":" and the split operation in auditlog/middleware.py considers a comma "," for splitting purposes. The way I see it, and I might be wrong, is to include an alternative split method considering ":" for cases when we do have ports in X-Forwarded-For value. Does it make sense?

seems that you are right and your proposed solution seems good.

we need to handle this in the `middleware.py`
https://github.com/jazzband/django-auditlog/blob/a00d2c227fad52e06d865037406d5483d0616b77/auditlog/middleware.py#L17-L22

We also need to add a proper test for this change.

Would you like to prepare the patch?

**Comment by alieh-rymasheuski** [comment]
Is this related to #238?

**Comment by hramezani** [comment]
Yes, seems so. it seems the fix is also prepared in https://github.com/jazzband/django-auditlog/issues/238#issuecomment-827548598.

So, we just need to make a patch out of it.
